File: group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = {}

        for s in strs:
            count = [0] * 26
            for c in s:
                count[ord(c) - ord("a")] += 1
            unique_key = tuple(count)
            if unique_key not in res:
                res[unique_key] = []
            res[unique_key].append(s)
        return list(res.values())

File: groupAnagrams.js
class Solution {
  /**
   * @param {string[]} strs
   * @return {string[][]}
   */
  groupAnagrams(strs) {
    let res = {};

    for (const s of strs) {
      let count = new Array(26).fill(0);
      for (const c of s) {
        count[c.charCodeAt(0) - "a".charCodeAt(0)]++;
      }
      const uniqueKey = count.join(",");
      if (!(uniqueKey in res)) {
        res[uniqueKey] = [];
      }
      res[uniqueKey].push(s);
    }
    return Object.values(res);
  }
}
